### PR TITLE
memdocstore: support unordered lists

### PR DESCRIPTION
**Pull request by jba**

### CODE REVIEW

File: internal/docstore/memdocstore/mem.go [hunk 1]
```diff
@@ -123,7 +130,22 @@ func (c *collection) ErrorCode(err error) gcerr.ErrorCode {
 // RunActions implements driver.RunActions.
 func (c *collection) RunActions(ctx context.Context, actions []*driver.Action, unordered bool) driver.ActionListError {
 	if unordered {
-		panic("unordered unimplemented")
+		errs := make([]error, len(actions))
+		for i, a := range actions {
+			i := i
+			a := a
+			go func() { errs[i] = c.runAction(ctx, a) }()
+		}
```
Comment (vangent) [inline internal/docstore/memdocstore/mem.go]: What prevents this from not waiting for all the goroutines to finish?

Comment (jba) [reply internal/docstore/memdocstore/mem.go]: Oops... I'm working a little too fast. Done.

**Comment by jba** [comment]
PTAL.

File: internal/docstore/memdocstore/mem.go (hunk 1 above)
Comment (vangent) [reply internal/docstore/memdocstore/mem.go]: As was your code...

**Pull request merged**